server/tserver/src/main/java/org/apache/accumulo/tserver/ThriftClientHandler.java

```diff
@@ -976,17 +1008,31 @@ public void update(TInfo tinfo, TCredentials credentials, TKeyExtent tkeyExtent,
         // Instead of always looping on true, skip completely when durability is NONE.
         while (durability != Durability.NONE) {
           try {
-            try (TraceScope wal = Trace.startSpan("wal")) {
+            Span span2 = tracer.spanBuilder("ThriftClientHandler::update::wal").startSpan();
+            try (Scope scope = span2.makeCurrent()) {
               server.logger.log(session, mutation, durability);
+            } catch (Exception e) {
+              span2.recordException(e, Attributes.builder().put("exception.message", e.getMessage())
+                  .put("exception.escaped", true).build());
+              throw e;
+            } finally {
+              span2.end();
             }
             break;
           } catch (IOException ex) {
             log.warn("Error writing mutations to log", ex);
           }
         }
 
-        try (TraceScope commit = Trace.startSpan("commit")) {
+        Span span3 = tracer.spanBuilder("ThriftClientHandler::update::commit").startSpan();
```

Comment:
These span builders on the server side could specify the span kind as SERVER. Also, I'm not sure where this happens, but shouldn't these set the parent to the span information incoming from the client side of the RPC?

Reply:
Take a look at TraceUtil.wrapService(). This method is called to wrap all of the Thrift servers and it takes the TInfo coming across the wire and creates a Context object from it and sets it as the current context for the call.